pkg/state/state_nv_peer.go

```diff
@@ -159,3 +172,36 @@ func (s *stateNVPeer) getManifestObjects(
 	log.V(consts.LogLevelDebug).Info("Rendered", "objects:", objs)
 	return objs, nil
 }
+
+// shouldIgnoreStatus check if nvPeerMem DS status should be ignored
+// Workaround to support switching from nv_peer_mem module which is managed by network-operator
+// to nvidia_peermem module which is a part of the GPU driver starting from v465.
+// We should ignore nvPeerMem status in case all nodes in a cluster have GPU driver version > 465
+// if some nodes still using GPU driver < 465 or there are no nodes with GPU driver, we should continue sync attempts
+// we don't remove DS, just ignore its status. This is required to be able to recovery nv-peer-mem in situations when
+// GPU driver version was downgraded or when we have an environment with mixed versions of the GPU drivers
+// and node with older version appear after node with never.
+// nvPeerMem config should be removed from NicClusterPolicy explicitly to remove DS
+func (s *stateNVPeer) shouldIgnoreStatus(nodeInfo nodeinfo.Provider) bool {
```

Comment:
maybe:

`cudaDriverNewer(cudaVer int)`

Then move the comment to where this is called ?

Reply:
I think that `shouldIgnoreStatus` function name is accurate because it describes what the function intends to do.
This logic is required to workaround the specific nv_peer_mem upgrade problem, so for me, it looks like a good idea to isolate workaround logic in a separate function and not to put it in the main `Sync` function. This will simplify this logic removal if required in the future.

Reply:
i see, OK